README.md

```diff
@@ -1,358 +1,302 @@
+option.fromNullable(null); // => option.none
+option.fromNullable('value'); // => option.some('value')
 
-#### <a name="get-options"></a>Get Value from an Option
-
-When your value is encapsulated into an Option, you sometimes want to retrieve it and opt-out of the functional paradigm. For example if you want to give it to the outside world, in a graphql resolver.
-
-##### Get the value or <null | undefined>
+// Build from a predicate
+const isEven = (number) => number % 2 === 0;
 
-Easiest way is to use the `toNullable` or `toUndefined` helpers. They are pretty straightforward, if your Option contains a value, you get it, otherwise you get null or undefined:
+option.fromPredicate(isEven)(3); // => option.none
+option.fromPredicate(isEven)(4); // => option.some(4)
 
-```typescript
-const noneValue = option.none;
-const someValue = option.of("value");
+// Convert from another type (eg. Either)
+const leftEither = either.left('whatever');
+const rightEither = either.right('value');
 
-option.toUndefined(noneValue); // undefined
-option.toUndefined(someValue); // "value"
-option.toNullable(noneValue); // null
-option.toNullable(someValue); // "value"
+option.fromEither(leftEither); // => option.none
+option.fromEither(rightEither); // => option.some('value')
 ```
 
-##### Get the value with a default
+#### <a name="option-getters"></a>Extract inner value
 
-You can use one of the following helper to retrieve your value or have a default if your `Option` is in `none` state.
-
-`getOrElse` takes a function as parameter that returns the default value for `none`. Please note the default value must have the same type as your initial Option. Eg: `getOrElse` on an `Option<number>` must return a `number`. If you want to return another type, you can use `getOrElseW`.
-
-```typescript
+```ts
 const noneValue = option.none;
-const someValue = option.of("value");
+const someValue = option.of('value');
 
-option.getOrElse(() => "default")(noneValue); // "default"
-option.getOrElse(() => "default")(someValue); // "value"
+// Convert Option<T> to T | undefined
+option.toUndefined(noneValue); // => undefined
+option.toUndefined(someValue); // => 'value'
 
-option.getOrElseW(() => 3)(noneValue); // 3
-```
+// Convert Option<T> to T | null
+option.toNullable(noneValue); // => null
+option.toNullable(someValue); // => 'value'
 
-##### Compute and get the value
+// Convert Option<T> with a default value
+option.getOrElse(() => 'default')(noneValue); // => 'default'
+option.getOrElse(() => 'default')(someValue); // => 'value'
 
-The last way to get your value is `match` and it allows you to compute before returning it.
-it takes a two functions, the first one is executed if your Option is `none`, the second one if your `Option` contains some value.
+// Convert Option<T> to T | U with a default value of type U
+option.getOrElseW(() => 3)(noneValue); // => 3: string | number
 
-```typescript
+// Apply a different function on None/Some(...)
 const noneValue = option.none;
```

Comment:
I would update the examples above so they use these two options instead of creating them twice. This would save a bit of space in the file